src/pal/src/loader/module.cpp

```diff
@@ -1349,8 +1349,8 @@ static HMODULE LOADLoadLibrary(LPCSTR ShortAsciiName, BOOL fDynamic)
     {
 #if defined(__APPLE__)
         ShortAsciiName = "libc.dylib";
-#elif defined(__FreeBSD__)
-        ShortAsciiName = "libc.so";
+#elif defined(__FreeBSD__) && defined(FREEBSD_LIBC)
```

Comment:
If we want to check that FREEBSD_LIBC is defined, it shouldn't fall down to the Linux case, it should error.